dipy/sims/voxel.py

```diff
@@ -188,7 +190,130 @@ def sticks_and_ball(gtab, d=0.0015, S0=1., angles=[(0, 0), (90, 0)],
     return S, sticks
 
 
-def single_tensor(gtab, S0=1., evals=None, evecs=None, snr=None):
+def callaghan_perpendicular(q, radius):
+    r""" Calculates the perpendicular diffusion signal E(q) in a cylinder of
+    radius R using the Soderman model [1]_. Assumes that the pulse length
+    is infinitely short and the diffusion time is infinitely long.
+
+    Parameters
+    ----------
+    q : array, shape (N,)
+        q-space value in 1/mm
+    radius : float
+        cylinder radius in mm
+
+    Returns
+    -------
+    E : array, shape (N,)
+        signal attenuation
+
+    References
+    ----------
+    .. [1]_ Söderman, Olle, and Bengt Jönsson. "Restricted diffusion in
+            cylindrical geometry." Journal of Magnetic Resonance, Series A
+            117.1 (1995): 94-97.
+    """
+    # Eq. [6] in the paper
+    E = ((2 * jn(1, 2 * np.pi * q * radius)) ** 2 /
+         (2 * np.pi * q * radius) ** 2
+         )
+    return E
+
+
+def gaussian_parallel(q, tau, D=0.7e-3):
+    r""" Calculates the parallel Gaussian diffusion signal.
+
+    Parameters
+    ----------
+    q : array, shape (N,)
+        q-space value in 1/mm
+    tau : float
+        diffusion time in s
+    D : float
+        diffusion constant
+
+    Returns
+    -------
+    E : array, shape (N,)
+        signal attenuation
+    """
+    return np.exp(-(2 * np.pi * q) ** 2 * tau * D)
+
+
+def cylinders_and_ball_soderman(gtab, tau, radii=[5e-3, 5e-3], D=0.7e-3,
+                                S0=1., angles=[(0, 0), (90, 0)],
+                                fractions=[35, 35], snr=20):
+    r""" Calculates the three-dimensional signal attenuation E(q) originating
+    from within a cylinder of radius R using the Soderman approximation [1]_.
+    The diffusion signal is assumed to be separable perpendicular and parallel
+    to the cylinder axis [2]_.
+    This function is basically an extension of the ball and stick model.
+    Setting the radius to zero makes them equivalent.
+
+    Parameters
+    ----------
+    gtab : GradientTable
+        Signal measurement directions.
+    tau : float
+        diffusion time in s
+    radii : float
+        cylinder radius in mm
+    D : float
+        diffusion constant
+    S0 : float
+        Unweighted signal value.
+    angles : array (K,2) or (K, 3)
+        List of K polar angles (in degrees) for the sticks or array of K
+        sticks as unit vectors.
+    direction : array (3)
+        direction of the axis of the cylinder
+    fractions : float
+        Percentage of each stick.  Remainder to 100 specifies isotropic
+        component.
+    snr : float
+        Signal to noise ratio, assuming Rician noise.  If set to None, no
+        noise is added.
+
+    Returns
+    -------
+    E : array, shape (N,)
+        signal attenuation
+
+    References
+    ----------
+    .. [1]_ Söderman, Olle, and Bengt Jönsson. "Restricted diffusion in
```

Comment:
Just a question, but why do you automatically put S0 to an arbitrary value that is not 1?
I saw it sometimes as 150 or 100 for default. Why not by default set it to 1 and keep the signal attenuation convention instead of unnormalized signal?